src/Api/Auth/Controllers/AccountsController.cs

```diff
@@ -939,6 +939,29 @@ public async Task PutUpdateTempPasswordAsync([FromBody] UpdateTempPasswordReques
         throw new BadRequestException(ModelState);
     }
 
+    [HttpPut("update-tde-offboarding-password")]
+    public async Task PutUpdateTdePasswordAsync([FromBody] UpdateTdeOffboardingPasswordRequestModel model)
```

Comment:
I think we need some validation here to prevent misuse of this endpoint. Like making sure that this request is coming from the intended device and that the organization actually is offboarding from TDE.

Reply:
Added validation and also unit tests for at least some of the validation. I'm not sure if all of it is required.